src/yunohost/backup.py

```diff
@@ -483,83 +891,299 @@ def _is_hook_in_backup(h):
-                env_dict["YNH_APP_INSTANCE_NAME"] = app_instance_name
-                env_dict["YNH_APP_INSTANCE_NUMBER"] = str(app_instance_nb)
-                env_dict["YNH_APP_BACKUP_DIR"] = tmp_app_bkp_dir
-
-                # Execute app restore script
-                hook_exec(app_script, args=[tmp_app_bkp_dir, app_instance_name],
-                          raise_on_error=True, chdir=tmp_app_bkp_dir, env=env_dict)
-            except:
-                logger.exception(m18n.n('restore_app_failed', app=app_instance_name))
+            # Cleaning app directory
+            shutil.rmtree(app_setting_path, ignore_errors=True)
+
+            # TODO Cleaning app hooks
+        else:
+            self.result['apps'].append(app_instance_name)
+        finally:
+            # Cleaning temporary scripts directory
+            shutil.rmtree(tmp_script_dir, ignore_errors=True)
+
+    def clean(self, retcode=0):
+        if self.result['apps']:
+            # Quickfix: the old app_ssowatconf(auth) instruction failed due to
+            # ldap restore hooks
+            os.system('sudo yunohost app ssowatconf')
+        ret = hook_callback('post_backup_restore', args=[self.work_dir,
+                                                         retcode])
+        if ret['failed']:
+            logger.warning(m18n.n('restore_cleaning_failed'))
+
+        if os.path.ismount(self.work_dir):
+            ret = subprocess.call(['umount', self.work_dir])
+            if ret != 0:
+                logger.warning(m18n.n('restore_cleaning_failed'))
+            else:
+                filesystem.rm(self.work_dir, True, True)
+
+    def _read_info_files(self):
+        # Retrieve backup info
+        info_file = os.path.join(self.work_dir, "info.json")
+        try:
+            with open(info_file, 'r') as f:
+                self.info = json.load(f)
+        except IOError:
+            logger.debug("unable to load '%s'", info_file, exc_info=1)
+            raise MoulinetteError(errno.EIO, m18n.n('backup_invalid_archive'))
+        else:
+            logger.debug("restoring from backup '%s' created on %s", self.name,
+                         time.ctime(self.info['created_at']))
 
-                app_script = tmp_app_dir + '/settings/scripts/remove'
+    def _get_env_var(self):
+        """ Define environment variable for hooks call """
+        env_var = {}
+        env_var['YNH_BACKUP_DIR'] = self.work_dir
+        env_var['YNH_BACKUP_CSV'] = os.path.join(self.work_dir, "backup.csv")
+        return env_var
 
-                # Setup environment for remove script
-                env_dict_remove = {}
-                env_dict_remove["YNH_APP_ID"] = app_id
-                env_dict_remove["YNH_APP_INSTANCE_NAME"] = app_instance_name
-                env_dict_remove["YNH_APP_INSTANCE_NUMBER"] = str(app_instance_nb)
+def backup_create(name=None, description=None, output_directory=None,
+                  no_compress=False, ignore_hooks=False, hooks=[],
```

Comment:
I'd really like if we could rename this `ignore_hooks` and `hooks` terminology into something like "system". The "hook" notion is really ambiguous and obscure for random users, and even as a dev it's confusing because "app scripts" are sometimes referred to as "hooks" (or "app hooks"). 

Also, seeing something like 

> yunohost backup create --hooks data --apps owncloud 
> yunohost backup create --hooks conf --apps owncloud

I would interpret it as "this is probably to backup only owncloud's data or owncloud's conf", whereas

> yunohost backup create --system data --apps owncloud 
> yunohost backup create --system conf --apps owncloud

is much clearer imho.